### PR TITLE
Modernize linux CI

### PR DESCRIPTION
**Pull request by juan-lunarg**
- Test Ubuntu-22.04
- Use ccache
- Ensure backwards compatibility with Ubuntu-20.04

### CODE REVIEW

**Review by ncesario-lunarg — APPROVED** [review]

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -16,51 +16,72 @@ permissions: read-all
...
+            - uses: actions/checkout@v3
+            - uses: lukka/get-cmake@latest  
+            - uses: actions/setup-python@v4
               with:
                   python-version: '3.7'
-            - name: Install Ubuntu Package Dependencies
-              run: |
-                  sudo apt-get -qq update
-                  sudo apt-get install -y clang-6.0
-            - name: Install GoogleTest
```
Comment (arcady-lunarg) [inline .github/workflows/continuous_integration.yml:35]: You still need to install (check out) googletest or else you're not running most of the test suite.

**Review by arcady-lunarg — CHANGES_REQUESTED** [review]
I think the approach to testing backwards compatibility is the right one, but removing the somewhat inaccurately named "install googletest" step gets rid of most of the test coverage we have.

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (juan-lunarg) [reply .github/workflows/continuous_integration.yml:35]: I figured that's what update_glslang_sources.py did. Sorry for the misunderstanding

**Review by arcady-lunarg — APPROVED** [review]
LGTM, thanks.

**Pull request merged**